admin/controllers/pupil.js

```diff
@@ -257,6 +261,28 @@ const postEditPupil = async (req, res, next) => {
   res.redirect(`/school/pupil-register/lastName/true?hl=${pupilId}`)
 }
 
+const getPrintPins = async (req, res, next) => {
```

Comment:
I think Pupil is large enough.  Now might be a good time to start splitting functionality.  How about creating a PupilPrintPinController?

Reply:
@jon-shipley print pin sounds too dedicated. How about one for generate pins instead?
Thoughts?

Reply:
How about pupil pin controller then?  To deal with generating, and printing for sure.  Maybe viewing?

Reply:
Yea I'll start refactoring

Reply:
Good to see this happening